C/common/readingset_circularbuffer.cpp

```diff
@@ -57,75 +65,26 @@ void ReadingSetCircularBuffer::insert(ReadingSet* readings)
  */
 void ReadingSetCircularBuffer::appendReadingSet(const std::vector<Reading *>& readings)
 {
+	bool isBufferFull = isFull();
 	lock_guard<mutex> guard(m_mutex);
 	//Check if there is space available to insert a new ReadingSet
-	if (isBufferFull())
+	if (isBufferFull)
```

Comment:
The buffer fullness status can change between isFull() call above and current line in multi-threading setting.